src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpEncapsulateField.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Roslyn.Test.Utilities;
+using Roslyn.VisualStudio.IntegrationTests.Extensions;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.CSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpEncapsulateField : AbstractEditorTest
+    {
+        public CSharpEncapsulateField(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(CSharpEncapsulateField))
+        {
+        }
+
+        private const string TestSource = @"
+namespace myNamespace
+{
+    class Program
+    {
+        private static int? $$param = 0;
+        static void Main(string[] args)
+        {
+            param = 80;
+        }
+    }
+}";
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        [Fact, Trait(Traits.Feature, Traits.Features.EncapsulateField)]
+        public void EncapsulateThroughCommand()
+        {
+            SetUpEditor(TestSource);
+
+            var dialog = VisualStudio.Instance.EncapsulateField;
```

Comment:
As mentioned below, this would be much less weird if it was `EncapsulateField.Invoke()` followed by `PreviewChanges.VerifyOpen`, etc.